src/Response.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Utopia\Fetch;
+
+/**
+ * Response class
+ * This class is used to represent the response from the server
+ * @package Utopia\Fetch
+ */
+class Response
+{
+    /**
+     * Response Body
+     *
+     * @var string
+     */
+    private string $body;
+    /**
+     * Response Headers
+     *
+     * @var array<string, string>
+     */
+    private array $headers;
+    /**
+     * Response Status Code
+     *
+     * @var int
+     */
+    private int $statusCode;
+    /**
+     * Response Method
+     *
+     * @var string
+     */
+    private string $method;
+    /**
+     * Response URL
+     *
+     * @var string
+     */
+    private string $url;
+    /**
+     * Response OK
+     *
+     * @var bool
+     */
+    private bool $ok;
```

Comment:
Why do we need a property for this ? 

In the isOk method we can just do `return $statusCode >= 200 && $statusCode < 300`; 

However this is also not the right check.. 300 series codes are also valid and can indicate a successful response.. 

I also dont think isOK should be part of the library. The reason being isOk depends on the application using the fetch library...

For my application 400 errors could mean a successful response... Its just that it was a bad request... So the isOK check should be left to the consumer of the library and can be removed from here .

Reply:
sure, I have removed the isOk check now